build/build.go

```diff
@@ -578,6 +580,21 @@ func toSolveOpt(ctx context.Context, node builder.Node, multiDriver bool, opt Op
 		}
 	}
 
+	if len(opt.Attests) > 0 {
+		if !bopts.LLBCaps.Contains(apicaps.CapID("exporter.image.attestations")) {
+			return nil, nil, errors.Errorf("attestations are not supported by the current buildkitd")
+		}
+		for k, v := range opt.Attests {
+			if v == nil {
+				continue
+			}
+			so.FrontendAttrs[k] = *v
+		}
+	}
+	if _, ok := opt.Attests["attest:provenance"]; !ok {
+		so.FrontendAttrs["attest:provenance"] = "mode=min,inline-only=true"
+	}
```

Comment:
Quick note to avoid confusion: we need `opt.Attests` to have some way of signalling a value that was explicitly disabled - we do that with `nil`.

This needs to be done here, so that we can ensure to only check the caps if we want something other than the default - otherwise, we want the `mode=min,inline-only=true` provenance to be attached on a best-effort basis, since the user hasn't explicitly opted in.